functions/src/import/config.ts

```diff
@@ -0,0 +1,204 @@
+import { Command } from 'commander'
+import * as inquirer from 'inquirer'
+import { MeilisearchConfig } from '../types'
+
+const FIRESTORE_VALID_CHARACTERS = /^[^/]+$/
+const FIRESTORE_COLLECTION_NAME_MAX_CHARS = 6144
+const PROJECT_ID_MAX_CHARS = 30
+const MEILISEARCH_VALID_CHARACTERS = /^[a-zA-Z-_0-9,]*$/
+const MEILISEARCH_UID_MAX_CHARS = 6144
+
+// Create a new interface to manage the arguments from the non-interactive command line.
+const program = new Command()
+
+program
+  .name('firestore-meilisearch')
+  .option(
+    '--non-interactive',
+    'Parse all input from command line flags instead of prompting the caller.',
+    false
+  )
+  .option(
+    '-P, --project <project>',
+    'Firebase Project ID for project containing the Cloud Firestore database.'
+  )
+  .option(
+    '-s, --source-collection-path <source-collection-path>',
+    'The path of the Cloud Firestore Collection to import from. (This may, or may not, be the same Collection for which you plan to mirror changes.)'
+  )
+  .option(
+    '-q, --query-collection-group [true|false]',
+    "Use 'true' for a collection group query, otherwise a collection query is performed."
+  )
+  .option(
+    '-i, --index <index>',
+    "The Uid of the index in Meilisearch to import to. (An index will be created if it doesn't already exist.)"
+  )
+  .option(
+    '-b, --batch-size [batch-size]',
+    'Number of documents to stream into Meilisearch at once.',
+    value => parseInt(value, 10),
+    1000
+  )
+  .option(
+    '-H, --host <host>',
+    'The Host of your Meilisearch database. Example: http://localhost:7700.'
+  )
+  .option(
+    '-a, --api-key <api-key>',
+    'The Meilisearch API key with permission to perform actions on indexes. Both the private key and the master key are valid choices but we strongly recommend using the private key for security purposes.'
+  )
+
+const validateInput = (
+  value: string,
+  name: string,
+  regex: RegExp,
+  sizeLimit: number
+) => {
+  if (!value || value === '' || !value.trim()) {
+    return `Please supply a ${name}`
+  }
+  if (value.length > sizeLimit) {
+    return `${name} must be at most ${sizeLimit} characters long`
+  }
+  if (!value.match(regex)) {
+    return `The ${name} does not match the regular expression provided`
+  }
+  return true
+}
+
+const validateBatchSize = (value: string) => {
+  return parseInt(value, 10) > 0
+}
```

Comment:
Contrary to `valdateInput` this function does not return a string explaining the issue, is this what you want?

Reply:
Yes, this function is just to parse the value into a `int` but I could check if the user doesn't try to put some string for batchsize instead of int but I hope they doesn't

Reply:
I would check! Never thrust the user